enhancements/kubelet/master-bootstrap-credentials.md

```diff
@@ -0,0 +1,120 @@
+---
+title: master-bootstrap-credentials
+authors:
+  - "@deads2k"
+reviewers:
+  - "@rphillips"
+approvers:
+  - "@rphillips"
+creation-date: 2020-01-31
+last-updated: 2020-01-31
+status: implementable
+see-also:
+  - https://bugzilla.redhat.com/show_bug.cgi?id=1693951
+replaces:
+superseded-by:
+---
+
+# Master Bootstrap Credentials
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+## Summary
+
+Update the master bootstrap credentials used to request the initial CSRs for a kubelet.
+Requirements for such a credential:
+ 1. long lived.  It should not expire on short time frames.
+ 2. revokable.  We must be able to revoke this credential if it is compromised.
+
+## Motivation
+
+In order to scale masters horizontally, replace masters, and shutdown and restore clusters,
+we need to have master bootstrap credentials that last longer.
+On workers, we use service account credentials
+
+### Goals
+
+Make it possible to
+ 1. scale masters horizontally
+ 2. replace masters
+ 3. shutdown and restore clusters
+
+### Non-Goals
+
+Actually perform the scaling.
+We need to remove this particular impediment, but the rest of the effort is out of scope.
+
+## Proposal
+
+ 1. In the installer, place the bundle from `kubelet-bootstrap-kubeconfig-ca-bundle` into openshift-config
+    called `configmap/kubelet-bootstrap-kubeconfig`.
+ 2. In the installer, sign the `kubelet-client` for ten years.  This is the actual bootstrap credential.
+ 3. In the cluster-kube-apiserver-operator, union in the content from `configmap/kubelet-bootstrap-kubeconfig`.
```

Comment:
I think this means "union it into KAS client CA bundle" but could be said explicitly.